### PR TITLE
1317: Update ios sdk

### PR DESCRIPTION
**Pull request by f1sh1918**
### Short description

Starting April 29, 2024, all iOS and iPadOS apps must be built with the iOS 17
SDK or later, included in Xcode 15 or later, in order to be uploaded to App
Store Connect or submitted for distribution.

### Proposed changes

- update to xcode 15.2
- update cocoapod version
- update gems
- update pods

### Side effects

- hopefully none, please test the app properly on ios

### Resolved issues

Fixes: #1317

### CODE REVIEW

File: .circleci/src/jobs/build_ios.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 macos:
-  xcode: 14.3.1
+  xcode: 15.2.0
```
Comment (ztefanie) [inline .circleci/src/jobs/build_ios.yml:2]: See this conversation: https://github.com/digitalfabrik/entitlementcard/pull/1248/files#r1449125363

Please check this upgrade does not break the CD

**Review by ztefanie — COMMENTED** [review]
Could not test. I get this error, when trying `pod install`
`[!] Unable to find a specification for MapLibreAnnotationExtension (~> 0.0.1-beta.2) depended upon by maplibre_gl`
Deleted the ~/.cocoapods folder and set it up again, but the error persists.

**Comment by f1sh1918** [comment]
> Could not test. I get this error, when trying `pod install`
> `[!] Unable to find a specification for MapLibreAnnotationExtension (~> 0.0.1-beta.2) depended upon by maplibre_gl`
> Deleted the ~/.cocoapods folder and set it up again, but the error persists. 

Did you Update cocoapods? It may rely on a local cocoapod installation

**Comment by ztefanie** [comment]
tried again and fixed some errors, now i get:

Failed to execute app-toolbelt. Exit code 0: (node:1690) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

when executing `fvm flutter pub run build_runner build --define "df_build_config=name=bayern"`

uninstalled the toolbelt and installed a fresh new version.

File: .circleci/src/jobs/build_ios.yml (hunk 1 above)
Comment (f1sh1918) [reply .circleci/src/jobs/build_ios.yml:2]: yes thats why i skip 15.0.1 :)

**Comment by f1sh1918** [comment]
> tried again and fixed some errors, now i get:
> 
> Failed to execute app-toolbelt. Exit code 0: (node:1690) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead. (Use `node --trace-deprecation ...` to show where the warning was created)
> 
> when executing `fvm flutter pub run build_runner build --define "df_build_config=name=bayern"`
> 
> uninstalled the toolbelt and installed a fresh new version.

which version of app-toolbelt do you use now?
I don't think this issue is connected to my pr

**Review by ztefanie — APPROVED** [review]
Thanks for this update 🚀 

Tested Nuernberg and Bayern, works as expected. Could not test scanning QR codes as I did not find a way how to pass an image to simulator.

**Comment by f1sh1918** [comment]
> Thanks for this update 🚀
> 
> Tested Nuernberg and Bayern, works as expected. Could not test scanning QR codes as I did not find a way how to pass an image to simulator.

you can not test scanning with ios simulator. you can only activate cards via deeplinking @ztefanie

**Pull request merged**